server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -460,42 +461,54 @@ private static long jitter(long ms) {
     }
 
     @Override
-    public List<TKeyExtent> bulkImport(TInfo tinfo, TCredentials credentials, long tid,
-        Map<TKeyExtent,Map<String,MapFileInfo>> files, boolean setTime)
+    public List<TKeyExtent> bulkImport(TInfo tinfo, TCredentials credentials, final long tid,
+        final Map<TKeyExtent,Map<String,MapFileInfo>> files, final boolean setTime)
         throws ThriftSecurityException {
 
       if (!security.canPerformSystemActions(credentials))
         throw new ThriftSecurityException(credentials.getPrincipal(),
             SecurityErrorCode.PERMISSION_DENIED);
 
-      List<TKeyExtent> failures = new ArrayList<>();
+      try {
+        return watcher.run(Constants.BULK_ARBITRATOR_TYPE, tid, new Callable<List<TKeyExtent>>() {
 
-      for (Entry<TKeyExtent,Map<String,MapFileInfo>> entry : files.entrySet()) {
-        TKeyExtent tke = entry.getKey();
-        Map<String,MapFileInfo> fileMap = entry.getValue();
-        Map<FileRef,MapFileInfo> fileRefMap = new HashMap<>();
-        for (Entry<String,MapFileInfo> mapping : fileMap.entrySet()) {
-          Path path = new Path(mapping.getKey());
-          FileSystem ns = fs.getVolumeByPath(path).getFileSystem();
-          path = ns.makeQualified(path);
-          fileRefMap.put(new FileRef(path.toString(), path), mapping.getValue());
-        }
+          @Override
+          public List<TKeyExtent> call() throws Exception {
```

Comment:
Was moving the Arbitrator thread here the fix?

Reply:
Yeap.   Also I forgot to mention that ignoring whitespace makes this diff really small.